File: src/app/models/event.ts
import { SafeResourceUrl } from "@angular/platform-browser";
import { EventLocation } from "./event-location";

export interface IEvent {
    id: number,
    categoryId: number,
    hostId: number,
    title: string,
    startingDate: string,
    endingDate: string,
    minimumParticipants: number,
    maximumParticipants: number,
    autoCancel: boolean,
    autoJoin: boolean,
    joinDeadline: string,
    fee: number,
    description: string,
    eventLocation: EventLocation,
    contactEmail: string,
    contactPhone: string,
    tags: string,
    backgroundImage: string
}

File: api/Models/Event.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace YonderfulApi.Models
{
    public class Event
    {
        [Key]
        public int Id { get; set; }
        public int CategoryId { get; set; }
        public int HostId { get; set; }
        public string Title { get; set; }
        public DateTime StartingDate { get; set; } 
        public DateTime EndingDate { get; set; }
        public int MinimumParticipants { get; set; }
        public int MaximumParticipants { get; set; }
        public bool Autocancel { get; set; }
        public bool AutoJoin { get; set; } 
        public DateTime JoinDeadline { get; set; }
        public int Fee { get; set; }
        public string Description { get; set; }
        public Location EventLocation { get; set; }
        public string ContactEmail { get; set; }
        public string ContactPhone { get; set; }
        public string Tags { get; set; }
        public int BackgroundId { get; set; }
    }
}